### PR TITLE
feat: suppress VersionMismatch warnings when listing wf runs

### PR DESCRIPTION
**Pull request by BenjaminMummery**
# The problem

VersionMismatch warnings are being spammed when dealing with multiple wf runs, specifically when listing wf runs or prompting the user to choose between runs.

VersionMismatch is raised when an older JSON representation of a workflow def is loaded in a new SDK version.

# This PR's solution

Suppresses VersionMismatch warnings when listing wf runs. 

This covers both the `orq list wf list` workflow, and the prompting (e.g. `orq wf logs` without specify a workflow).

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/internal/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

[ORQSDK-843]

[ORQSDK-843]: https://zapatacomputing.atlassian.net/browse/ORQSDK-843?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by BenjaminMummery** [comment]
If anyone knows a good way to automate tests for this give me a shout, I kept running into clashes between what pytest wanted to do with logged warnings and what our code wanted to do.

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 * `sdk.secrets.list()`, `sdk.secrets.get()`, `sdk.secrets.set()` and `sdk.secrets.delete()` now accept `workspace_id` parameter to specify secrets in particular workspace.
 * `auto` config inside studio will infer workspace and project IDs from studio instance.
 * Support for running tasks in Docker containers with custom images on Compute Engine.
+* VersionMismatch warnings are shown only when interacting with specific workflow runs, not while listing workflows.
```
Comment (jamesclark-Zapata) [inline CHANGELOG.md:20]: Make sure to put this in the new unreleased section after #176

**Comment by jamesclark-Zapata** [comment]
> If anyone knows a good way to automate tests for this give me a shout, I kept running into clashes between what pytest wanted to do with logged warnings and what our code wanted to do.

You might be able to set the `VersionMismatch` warning to be an exception, then the test passing is just running without a warning.
In the test code:
```
warnings.filterwarnings("error", category=exceptions.VersionMismatch)
```

File: src/orquestra/sdk/_base/cli/_dorq/_repos.py [hunk 2]
```diff
@@ -83,19 +83,22 @@ def list_wf_runs(
...
-                limit=limit,
-                max_age=max_age,
-                state=state,
-                workspace=workspace,
-                project=project,
-            )
-        except (ConnectionError, exceptions.UnauthorizedError):
-            raise
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", category=exceptions.VersionMismatch)
```
[the remark this answers is not in the document]
Comment (BenjaminMummery) [reply src/orquestra/sdk/_base/cli/_dorq/_repos.py]: Good point, I've moved the filtering down a level so it covers the API as well, which I think is closer to what we would like to happen.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,7 @@
 
 🔥 *Features*
 * Add .project property to WorkflowRun to get the info about workspace and project of running workflow
+* VersionMismatch warnings are shown only when interacting with specific workflow runs, not while listing workflows.
```
Comment (jamesclark-Zapata) [inline CHANGELOG.md:9]: **suggestion (non-blocking):** ‏Some formatting is nice in the markdown.
```suggestion
* `VersionMismatch` warnings are shown only when interacting with specific workflow runs, not while listing workflows.
* ```

**Review by jamesclark-Zapata — APPROVED** [review]

**Pull request merged**